### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by marcinbojko**

### CODE REVIEW

**Comment by genebean** [comment]
Sorry for leaving this sitting so long, I haven't had the time I used to to work on modules. I like some of the things you are doing here and we'll get them added in. Are you interested in working with me to tweak this a little?

**Comment by marcinbojko** [comment]
Sure. However I let myself to go deeper :) Your module is very interesting and well written. I was not able to repeat it on my own, but your ground work was excellent ;)

File: .scripts/foreman.home.lan-dev.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
```
Comment (genebean) [inline .scripts/foreman.home.lan-dev.sh]: Why is this included in this PR? I am not sure what its for other than your local testing.

File: manifests/install.pp [hunk 2]
```diff
@@ -15,12 +17,24 @@
     } # end Linux
 
     'Windows' : {
```
Comment (genebean) [inline manifests/install.pp]: I believe this can be done with simpler code by simply setting a the value for `$windows_package_source` to `undef` in the params file. By doing this you don't need the if statement but rather just the one that is in the "if" part now (not the "else" part).

File: manifests/params.pp [hunk 3]
```diff
@@ -5,9 +5,11 @@
   $manage_repo_zabbix     = false
 
   # install settings
-  $ensure_setting         = 'present'
+  $ensure_setting         = 'latest'
   $custom_require_linux   = undef
   $custom_require_windows = undef
+  $windows_package_name   = 'zabbix-agent3'
+  $windows_package_source = ''
```
Comment (genebean) [inline manifests/params.pp]: Per previous comment, change `''` to `undef`

File: manifests/params.pp [hunk 4]
```diff
@@ -5,9 +5,11 @@
   $manage_repo_zabbix     = false
 
   # install settings
-  $ensure_setting         = 'present'
+  $ensure_setting         = 'latest'
```
Comment (genebean) [inline manifests/params.pp]: I do not want this set to `latest`... users are welcome to override it locally if they desire.

File: README.md [hunk 5]
```diff
@@ -82,6 +82,13 @@ Default: undef
 Passed directly to require of package resource when on Windows  
 Default: undef
 
+#####`windows_package_name`
```
Comment (genebean) [inline README.md]: Lets change this to just `package_name` as its not really Windows specific. As part of that also update the comment to something like "The name of package to install. Defaults to `zabbix-agent`."

File: metadata.json [hunk 6]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "genebean-zabbixagent",
-  "version": "2.6.0",
+  "version": "2.6.1",
```
Comment (genebean) [inline metadata.json]: Please don't include this change. I will update the metadata when I do a release.

File: manifests/params.pp [hunk 7]
```diff
@@ -5,9 +5,11 @@
...
   $custom_require_linux   = undef
   $custom_require_windows = undef
+  $windows_package_name   = 'zabbix-agent3'
```
Comment (genebean) [inline manifests/params.pp]: Per previous comment, please update the name here to just `package_name` and a value of `zabbix-agent`.

File: manifests/install.pp [hunk 8]
```diff
@@ -2,7 +2,9 @@
 class zabbixagent::install (
   $ensure_setting         = $::zabbixagent::ensure_setting,
   $custom_require_linux   = $::zabbixagent::custom_require_linux,
-  $custom_require_windows = $::zabbixagent::custom_require_windows) {
+  $custom_require_windows = $::zabbixagent::custom_require_windows,
+  $windows_package_name   = $::zabbixagent::windows_package_name,
```
Comment (genebean) [inline manifests/install.pp]: With the changes to this param I've mentioned it could be used on the Linux resource name too... no biggie if you want me to add that myself later.

File: .scripts/foreman.home.lan-dev.sh (hunk 1 above)
Comment (marcinbojko) [inline .scripts/foreman.home.lan-dev.sh]: That was my first PR. Please forget it.

File: manifests/params.pp (hunk 4 above)
Comment (marcinbojko) [inline manifests/params.pp]: Agreed.

File: .scripts/foreman.home.lan-dev.sh (hunk 1 above)
Comment (genebean) [inline .scripts/foreman.home.lan-dev.sh]: No problem! As you work through this one and the other comments just adjust your code and re-push.

**Comment by marcinbojko** [comment]
@genebean - I think I should delete my PR. I am not a dev, I did a lots of changes in next (2.6.2) version (unpublish on github), and I have a really hard time to understand how to revert to your directions.

**Comment by genebean** [comment]
No need to delete unless you just want to! If what I am asking is too much then leave it on GH and just close the PR. I will use what you have posted here as a basis for updating the module as its good work. I'd also welcome any other changes you have going and am happy to help you with anything you are stuck on.

**Comment by marcinbojko** [comment]
No, I suppose your request are valid, but my skills are not good enough.
I just wanted to use different zabbix-agent (3) with specific source. The thing is, that current PR is just quite ancient, and my changes were much more deeper. So, I am looking for a best way to resolve this.
Maybe I'll try from fresh clone?

**Comment by genebean** [comment]
A from-scratch would be perfectly fine! In case it makes a difference, I am working with a co-worker on another PR right now too so you **might** want to hold off until its merged (likely this weekend or Monday). FWIW, I also work on the Chocolatey package and will be updating it very soon.

**Comment by marcinbojko** [comment]
Sure, I saw this PR (about zabbix 3.2), I can hold on.
I have working zabbix-agent3 if you would like to check, all we need to change would be probably paths to a files. 
https://github.com/marcinbojko/zabbix-agent

**Comment by genebean** [comment]
I'll definitely take a look at your package as it may well save me some time. The fine folks at Zabbix SIA finally responded to my PR on their package a little while back and added me as a maintainer so we'll actually be able to switch back to their GitHub soon... just thought you'd like to know.

**Comment by marcinbojko** [comment]
Cool! Before that, Windows agents were very rarely updated, so I had to do it on my own.
Both (puppet and choco packages) you've created were **VERY** helpful.

**Comment by genebean** [comment]
Thanks! I am going to hit close on this PR but will welcome your next one :)

**Comment by genebean** [comment]
BTW, the "Virus Total check link" on your package is a great looking addition

**Comment by marcinbojko** [comment]
All my packages have this, as I am using (sometimes) dropbox as source.
If you would like I'll add you to my myget source.

**Comment by genebean** [comment]
No thanks but I appreciate it!

**Comment by genebean** [comment]
@marcinbojko -- just wanted to let you know that the new `3.0.0` version has been pushed to the forge and that we plan on working on the Chocolatey module today. Feel free to start your new PR whenever you are ready. Thanks!

**Comment by marcinbojko** [comment]
Thanks!
